src/v/kafka/server/handlers/topics/types.cc

```diff
@@ -111,7 +111,14 @@ static std::optional<bool>
 get_bool_value(const config_map_t& config, std::string_view key) {
```

Comment:
Looks like not all bools are treated equally 😠 

```
 → ~/redpanda/vbuild/go/linux/amd64/bin/rpk topic create foo --config write.caching=TruE
TOPIC  STATUS
foo    OK

---

 → ~/redpanda/vbuild/go/linux/amd64/bin/rpk topic describe foo -a | grep write.caching
write.caching                         false       DEFAULT_CONFIG
```

Then the more interesting case. I apply the following path to rpk

```
diff --git a/src/go/rpk/pkg/cli/topic/create.go b/src/go/rpk/pkg/cli/topic/create.go
index 5e2ca5b410..6da0b15b89 100644
--- a/src/go/rpk/pkg/cli/topic/create.go
+++ b/src/go/rpk/pkg/cli/topic/create.go
@@ -82,6 +82,17 @@ the cleanup.policy=compact config option set.
 				reqTopic.ReplicationFactor = replicas
 				reqTopic.NumPartitions = partitions
 				reqTopic.Configs = reqConfigs
+
+				if topic == "foo" {
+					reqTopic.Configs = append(reqTopic.Configs, kmsg.NewCreateTopicsRequestTopicConfig())
+					reqTopic.Configs[len(reqTopic.Configs)-1].Name = "write.caching"
+					reqTopic.Configs[len(reqTopic.Configs)-1].Value = kmsg.StringPtr("foo")
+				} else if topic == "bar" {
+					reqTopic.Configs = append(reqTopic.Configs, kmsg.NewCreateTopicsRequestTopicConfig())
+					reqTopic.Configs[len(reqTopic.Configs)-1].Name = "write.caching"
+					reqTopic.Configs[len(reqTopic.Configs)-1].Value = kmsg.StringPtr("TruE")
+				}
+
 				req.Topics = append(req.Topics, reqTopic)
 			}
```

and run the following command 

```
 → task rpk:build && ~/redpanda/vbuild/go/linux/amd64/bin/rpk topic create foo bar baz
task: Task "rpk:build" is up to date
TOPIC  STATUS
baz    OK
bar    INVALID_CONFIG: Unsupported write caching configuration.
foo    INVALID_CONFIG: Unsupported write caching configuration.
 λ nv at dev in /home/nv/redpanda on git:create_topics_boolean_handling[!]
```

I'm not sure I understand what's going on. But in this later case both `foo` and `TruE` were rejected. But they are not rejected in the case above?